signer/src/bitcoin/validation.rs

```diff
@@ -360,6 +386,79 @@ impl DepositRequestReport {
     }
 }
 
+/// An enum for the confirmation status of a withdrawal request.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum WithdrawalRequestStatus {
+    /// We have a record of the withdrawal request transaction, and it has
+    /// been confirmed on the canonical Stacks blockchain. We have not
+    /// fulfilled the request. The integer is the height of the bitcoin
+    /// block anchoring the Stacks block that confirmed the withdrawal
+    /// request, and the block hash is the associated block hash of that
+    /// bitcoin block.
+    Confirmed(u64, BitcoinBlockHash),
+    /// We have a record of the withdrawal request being included as an
+    /// output in another bitcoin transaction that has been confirmed on
+    /// the canonical bitcoin blockchain.
+    Fulfilled(BitcoinTxId),
+    /// We have a record of the withdrawal request transaction, and it has
+    /// not been confirmed on the canonical Stacks blockchain.
+    ///
+    /// Usually we will almost certainly have a record of a withdrawal
+    /// request, and we require that the withdrawal transaction be
+    /// confirmed before we write it to our database. But the withdrawal
+    /// transaction can be affected by a bitcoin reorg, where it is no
+    /// longer confirmed on the canonical bitcoin blockchain. If this
+    /// happens when we query for the status then it will come back as
+    /// unconfirmed.
+    Unconfirmed,
+}
+
+/// A struct for the status report summary of a withdrawal request for use
+/// in validation.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct WithdrawalRequestReport {
+    /// The confirmation status of the withdrawal request transaction.
+    pub status: WithdrawalRequestStatus,
+    /// The request id generated by the smart contract when the
+    /// `initiate-withdrawal-request` public function was called.
+    pub request_id: u64,
+    /// The stacks transaction ID that lead to the creation of the
+    /// withdrawal request.
+    pub txid: StacksTxId,
+    /// Stacks block ID of the block that includes the transaction
+    /// associated with this withdrawal request.
+    pub block_hash: StacksBlockHash,
+    /// The amount of BTC, in sats, to withdraw.
+    pub amount: u64,
+    /// The max fee amount to use for the sBTC deposit transaction.
+    pub max_fee: u64,
+    /// The script_pubkey of the output.
+    pub script_pubkey: ScriptBuf,
+}
+
+impl WithdrawalRequestReport {
+    fn qualified_id(&self) -> QualifiedRequestId {
+        QualifiedRequestId {
+            block_hash: self.block_hash,
+            txid: self.txid,
+            request_id: self.request_id,
+        }
+    }
+    /// Validate that the deposit request is okay given the report.
```

Comment:
```suggestion
    /// Validate that the withdrawal request is okay given the report.
```

Reply:
fixed up in https://github.com/stacks-network/sbtc/pull/841/commits/4d70997616aa67eb9821bba3de938f3f7e8bc35e